network_analysis/configure-subnetwork-trace/src/main/java/com/esri/samples/configure_subnetwork_trace/ConfigureSubnetworkTraceController.java

```diff
@@ -0,0 +1,419 @@
+              .collect(Collectors.toList());
+          comparisonSourcesComboBox.getItems().addAll(comparisonSources);
+          comparisonSourcesComboBox.getSelectionModel().select(0);
+          // display the name of the comparison sources in the ComboBox
+          comparisonSourcesComboBox.setButtonCell(new ComparisonSourceListCell());
+          comparisonSourcesComboBox.setCellFactory(c -> new ComparisonSourceListCell());
+
+          // build the choice list for the comparison operators
+          comparisonOperatorsComboBox.getItems().addAll(UtilityAttributeComparisonOperator.values());
+          comparisonOperatorsComboBox.getSelectionModel().select(0);
+
+          // display the name of the comparison values in the ComboBox
+          comparisonValuesComboBox.setButtonCell(new CodedValueListCell());
+          comparisonValuesComboBox.setCellFactory(c -> new CodedValueListCell());
+
+          // create a default starting location
+          UtilityNetworkSource utilityNetworkSource =
+              utilityNetwork.getDefinition().getNetworkSource("Electric Distribution Device");
+          UtilityAssetGroup utilityAssetGroup = utilityNetworkSource.getAssetGroup("Circuit Breaker");
+          UtilityAssetType utilityAssetType = utilityAssetGroup.getAssetType("Three Phase");
+          startingLocation =
+              utilityNetwork.createElement(utilityAssetType, UUID.fromString("1CAF7740-0BF4-4113-8DB2-654E18800028"));
+
+          // set the terminal for the starting location. (For our case, we use the 'Load' terminal.)
+          List<UtilityTerminal> terminals = startingLocation.getAssetType().getTerminalConfiguration().getTerminals();
+          terminals.forEach(terminal -> {
+            if (terminal.getName().equals("Load")) {
+              startingTerminal = terminal;
+            }
+          });
+          startingLocation.setTerminal(startingTerminal);
+
+          // get a default trace configuration from a tier to update the UI
+          UtilityDomainNetwork utilityDomainNetwork =
+              utilityNetwork.getDefinition().getDomainNetwork("ElectricDistribution");
+          UtilityTier utilityTier = utilityDomainNetwork.getTier("Medium Voltage Radial");
+          utilityTraceConfiguration = utilityTier.getTraceConfiguration();
+
+          // save the default trace configuration to restore when the application is reset
+          initialUtilityTraceConfiguration = utilityTraceConfiguration;
+
+          // save the initial expression
+          initialExpression =
+              (UtilityTraceConditionalExpression) utilityTier.getTraceConfiguration().getTraversability().getBarriers();
+
+          // show the initial expression in the text area
+          traceConditionsTextArea.setText(expressionToString(initialExpression));
+
+          // set the traversability scope
+          utilityTier.getTraceConfiguration().getTraversability().setScope(UtilityTraversabilityScope.JUNCTIONS);
```

Comment:
I think you can replace this with the following since you stored it in a variable earlier
```suggestion
          utilityTraceConfiguration().getTraversability().setScope(UtilityTraversabilityScope.JUNCTIONS);
```